src/input/type_record.cc

```diff
@@ -359,6 +365,48 @@ Record &Record::has_obligatory_type_key() {
 }
 
 
+boost::shared_ptr<TypeBase> Record::make_instance(std::vector<ParameterPair> vec) const {
+	Record rec = Record(this->type_name(), this->data_->description_);
+	// Add parent Abstracts
+	for (auto it = data_->parent_ptr_.begin(); it != data_->parent_ptr_.end(); ++it) {
+		rec.derive_from( *(*it) );
+	}
+	// Set autoconversion key
+	if (data_->auto_conversion_key != "") rec.allow_auto_conversion(data_->auto_conversion_key);
+	// Copy keys
+	rec.copy_keys(*this);
+	// Replace keys of type Parameter
+	for (std::vector<Key>::iterator key_it=rec.data_->keys.begin(); key_it!=rec.data_->keys.end(); key_it++) {
+		if ( key_it->key_ != "TYPE" && typeid( *(key_it->type_) ) == typeid(Parameter) ) {
+			bool found = false;
+			for (std::vector<ParameterPair>::iterator vec_it=vec.begin(); vec_it!=vec.end(); vec_it++) {
+				if ( (*vec_it).first == key_it->key_ ) {
+					found = true;
+					key_it->type_ = (*vec_it).second;
+				}
+			}
+			if (!found) xprintf(Warn, "Parameterized key '%s' in make_instance method of '%s' Record wasn't replaced!\n",
+					key_it->key_.c_str(), this->type_name().c_str());
+		}
+	}
+	// Copy attributes
+	for (attribute_map::iterator it=attributes_->begin(); it!=attributes_->end(); it++) {
```

Comment:
Ok, still we can use assignment operator for the map:
 rec.attributes_ = std::make_shared(*attributes_).

I assume 
1) call of Record copy constructor (which in turn call TypeBase copy constructor)
    after that both original and new record attributes_ points to the same map.
2) use make_shared to make new pointer to copy of the map.